GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -8,10 +8,101 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
+    var allEpisodesinSeasons = arrOfSeasonsData
+    var arrOfSeasons: [[GOTEpisode]] {
+        guard let searchString = searchString else {
+            return allEpisodesinSeasons
+        }
+        guard searchString != "" else {
+            return allEpisodesinSeasons
+        }
+        let searchResults = filterForSearchResult(arr: GOTEpisode.allEpisodes, string: searchString)
+        let separateSeasons = makeArrofSeasons(allEp: searchResults, seasons: 7)
+        return separateSeasons
+        
+    }
+    var searchString: String? = nil {
+        didSet {
+            self.gOTTableView.reloadData()
+        }
+    }
+    
+    //MARK: -- TableView Methods
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        switch section % 2 {
+        case 0:
+            return arrOfSeasons[section].count
+        case 1:
+            return arrOfSeasons[section].count
+        default:
+            return GOTEpisode.allEpisodes.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "cellOne") as? cellOneTableViewCell else {
+            return UITableViewCell()
+        }
+        guard let cellTwo = tableView.dequeueReusableCell(withIdentifier: "cellTwo") as? cellTwoTableViewCell else {
+            return UITableViewCell()
+        }
+        let section = indexPath.section
+        let episodeArr = arrOfSeasons[section]
+        let theEpisode = episodeArr[indexPath.row]
+        switch indexPath.section % 2 {
```

Comment:
Instead of using a switch, maybe do an if else. That way you dont need a default case to return an empty cell.